pyparsing/diagram/__init__.py

```diff
@@ -424,3 +452,27 @@ def _to_diagram_element(
         )
     else:
         return ret
+
+
+# monkeypatch .create_diagram method onto ParserElement
+def _create_diagram(expr: pyparsing.ParserElement, output_html: Union[str, TextIO]):
```

Comment:
I think this might be better as a method on the original `ParserElement` class, which checks for an `ImportError` as you have done at the top. I prefer this to monkey patching, anyway.

Reply:
My intention behind monkeypatching is so that the core library can be easily vendored without having to include dependencies on other packages beside the stdlib.

Reply:
But if you do this, is that not sufficient? Then the dependencies are still optional.
```python
try:
    import pyparsing.diagram
except ImportError:
    raise Exception('You need to run `pip install pyparsing[diagrams]`')

Reply:
I got rid of the monkeypatch and added create_diagram to ParserElement as you suggested.